### ISSUE TITLE
Cannot update / delete schedules after upgrade to 3.0.1

### ISSUE THREAD

**Issue by gmergulhao** [aws-solutions/instance-scheduler-on-aws#562]
**Describe the bug**

Cannot update or delete stacks which handle schedules. Lambda no longer accepts property 'UseMaintenanceWindow'

After upgrading from 1.5.3 we noticed that one can no longer update or even successfully delete stacks which created schedules. This has been done so for about 5 years, according to [aws documentation](https://docs.aws.amazon.com/solutions/latest/instance-scheduler-on-aws/manage-schedules-using-infrastructure-as-code-iac.html)

The lambda InstanceSchedulerMain no longer accepts the property 'UseMaintenanceWindow', which differs from the documentation above, and returns error to cloudformation.
The same behaviour occurs when trying to delete the stack (which sends event delete to lambda) or even updating the stack to a new template version which does **not** include 'UseMaintenanceWindow'. The lambda apparently still checks all properties within 'OldResourceProperties' as it is part of the event.

**To Reproduce**

- setup aws-instance-scheduler on version 1.5.3
- Create a schedule using cloudformation according to the aws doc above
- update scheduler to 3.0.1
- Update / delete the stack

**Expected behavior**

- The lambda does not return error if a legacy property was passed within event on update / delete
- cfn-Stacks with existing schedules can be updated/deleted without problems

**information about the solution:**

- [ ] Version:(SO0030) instance-scheduler-on-aws v3.0.1

- [ ] Region: eu-central-1
- [ ] Was the solution modified from the version published on this repository? No
- [ ] If the answer to the previous question was yes, are the changes available on GitHub?
- [ ] Have you checked your [service quotas](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html) for
      the sevices this solution uses? Yes
- [ ] Were there any errors in the CloudWatch Logs?

       Unknown schedule property UseMaintenanceWindow, valid properties are dict_keys(['ServiceToken', 'NoStackPrefix', 'Name', 'Description', 'Timezone', 'Enforced', 'Hibernate', 'RetainRunning', 'StopNewInstances', 'SsmMaintenanceWindow', 'Metrics', 'OverrideStatus', 'Periods'])

**event of a request 'update' with the property 'UseMaintenanceWindow' only within 'OldResourceProperties'**
```
"ResourceProperties": {
        "ServiceToken": "arn:aws:lambda:eu-central-1:123123123:function:xyz-Instance-Scheduler-v140-InstanceSchedulerMain",
        "Timezone": "Europe/Berlin",
        "StopNewInstances": "false",
        "Periods": [
            {
                "WeekDays": "Mon-Sun",
                "Description": "Schedule for RDS Backup",
                "EndTime": "03:10",
                "BeginTime": "00:00"
            },
            {
                "WeekDays": "Mon-Fri",
                "Description": "Period 1",
                "EndTime": "10:00",
                "BeginTime": "09:00"
            }
        ],
        "Description": "Specific Time Window for Application: jean",
        "Enforced": "False",
        "Name": "jean-SpecificDefinitionRDS-3",
        "NoStackPrefix": "True"
    },
    "OldResourceProperties": {
        "ServiceToken": "arn:aws:lambda:eu-central-1:123123123:function:xyz-Instance-Scheduler-v140-InstanceSchedulerMain",
        "Timezone": "Europe/Berlin",
        "StopNewInstances": "false",
        "UseMaintenanceWindow": "True",
        "Periods": [
            {
                "WeekDays": "Mon-Sun",
                "Description": "Schedule for RDS Backup",
                "EndTime": "03:10",
                "BeginTime": "00:00"
            },
            {
                "WeekDays": "Mon-Fri",
                "Description": "Period 1",
                "EndTime": "10:00",
                "BeginTime": "09:00"
            }
        ],
        "Description": "Specific Time Window for Application:  jean",
        "Enforced": "False",
        "Name": "jean-SpecificDefinitionRDS-3",
        "NoStackPrefix": "True"
    }
```

**Additional context**

**Comment by CrypticCabub** [comment]
Thanks for reporting this. This is indeed a bug in the 3.0.x release of Instance Scheduler and will be fixed in our next patch release

**Comment by CrypticCabub** [comment]
Fixed in [v3.0.2](https://github.com/aws-solutions/instance-scheduler-on-aws/releases/tag/v3.0.2)
